src/main/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920PerFieldKnnVectorsFormat.java

```diff
@@ -22,7 +22,9 @@ public KNN920PerFieldKnnVectorsFormat(final Optional<MapperService> mapperServic
             Lucene92HnswVectorsFormat.DEFAULT_MAX_CONN,
             Lucene92HnswVectorsFormat.DEFAULT_BEAM_WIDTH,
             () -> new Lucene92HnswVectorsFormat(),
-            (maxConnm, beamWidth) -> new Lucene92HnswVectorsFormat(maxConnm, beamWidth)
+            (maxConnm, beamWidth) -> new Lucene92HnswVectorsFormat(maxConnm, beamWidth),
+            (maxConnm, beamWidth, confidenceInterval, bits, compress) -> new Lucene92HnswVectorsFormat(maxConnm, beamWidth)
```

Comment:
why we are updating the older codec? is it needed?

Reply:
It is forcing me to do that as part of initializing the constructor. Still working on that to see if we can skip it somehow. Vikash is also helping me here looking for alternate options.

Reply:
There are two way to avoid changing in base classes , 
1. Overload constructor in baseFormat Class and this is the other way ...

Here is how base class will looks like 

``
```
@AllArgsConstructor
@Log4j2
public abstract class BasePerFieldKnnVectorsFormat extends PerFieldKnnVectorsFormat {
    private final Optional<MapperService> mapperService;
    private final int defaultMaxConnections;
    private final int defaultBeamWidth;
    private final Supplier<KnnVectorsFormat> defaultFormatSupplier;
    private final BiFunction<Integer, Integer, KnnVectorsFormat> formatSupplier;
    private final Function<KNNVectorsFormatParams, KnnVectorsFormat> vectorsFormatSupplier;

    @Override
    public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
        var type = mapperService.map(service -> service.documentMapper().mappers().getMapper(field)).orElseThrow(
            () -> new IllegalStateException("No mapper for field [" + field + "]")
        ).fieldType(field);
        var params = type.getKnnMethodContext().getMethodComponentContext().getParameters();
        int maxConnections = getMaxConnections(params);
        int beamWidth = getBeamWidth(params);

        if (type.getKnnMethodContext().getKnnEngine() == KNNEngine.LUCENE
            && params != null
            && params.containsKey(METHOD_ENCODER_PARAMETER)) {
            KNNScalarQuantizedVectorsFormatParams knnScalarQuantizedVectorsFormatParams = new KNNScalarQuantizedVectorsFormatParams(
                params,
                defaultMaxConnections,
                defaultBeamWidth
            );
            if (knnScalarQuantizedVectorsFormatParams.validate(params)) {
                log.debug(
                    "Initialize KNN vector format for field [{}] with params [{}] = \"{}\", [{}] = \"{}\", [{}] = \"{}\", [{}] = \"{}\"",
                    field,
                    MAX_CONNECTIONS,
                    knnScalarQuantizedVectorsFormatParams.getMaxConnections(),
                    BEAM_WIDTH,
                    knnScalarQuantizedVectorsFormatParams.getBeamWidth(),
                    LUCENE_SQ_CONFIDENCE_INTERVAL,
                    knnScalarQuantizedVectorsFormatParams.getConfidenceInterval(),
                    LUCENE_SQ_BITS,
                    knnScalarQuantizedVectorsFormatParams.getBits()
                );
                return getScalarQuantizedVectorsFormatSupplier().apply(knnScalarQuantizedVectorsFormatParams);
            }
        }

        KNNVectorsFormatParams knnVectorsFormatParams = new KNNVectorsFormatParams(params, defaultMaxConnections, defaultBeamWidth);
        log.debug(
            "Initialize KNN vector format for field [{}] with params [max_connections] = \"{}\" and [beam_width] = \"{}\"",
            field,
            maxConnections,
            beamWidth,
            knnVectorsFormatParams.getMaxConnections(),
            knnVectorsFormatParams.getBeamWidth()
        );
        return vectorsFormatSupplier.apply(knnVectorsFormatParams);
    }

    protected Function<KNNScalarQuantizedVectorsFormatParams, KnnVectorsFormat> getScalarQuantizedVectorsFormatSupplier() {
        throw new UnsupportedOperationException("Scalar Quantized Vectors Format Supplier not supported");
    }

    private int getMaxConnections(Map<String, String> params) {
        // Implementation to get max connections from params or return defaultMaxConnections
        return defaultMaxConnections;
    }

    private int getBeamWidth(Map<String, String> params) {
        // Implementation to get beam width from params or return defaultBeamWidth
        return defaultBeamWidth;
    }
}
```
```
public class KNN990PerFieldKnnVectorsFormat extends BasePerFieldKnnVectorsFormat {
    private final Function<KNNScalarQuantizedVectorsFormatParams, KnnVectorsFormat> scalarQuantizedVectorsFormatSupplier;

    public KNN990PerFieldKnnVectorsFormat(final Optional<MapperService> mapperService) {
        super(
            mapperService,
            Lucene99HnswVectorsFormat.DEFAULT_MAX_CONN,
            Lucene99HnswVectorsFormat.DEFAULT_BEAM_WIDTH,
            () -> new Lucene99HnswVectorsFormat(),
            (maxConn, beamWidth) -> new Lucene99HnswVectorsFormat(maxConn, beamWidth),
            knnVectorsFormatParams -> new Lucene99HnswVectorsFormat(
                knnVectorsFormatParams.getMaxConnections(),
                knnVectorsFormatParams.getBeamWidth()
            )
        );
        this.scalarQuantizedVectorsFormatSupplier = knnScalarQuantizedVectorsFormatParams -> new Lucene99HnswScalarQuantizedVectorsFormat(
            knnScalarQuantizedVectorsFormatParams.getMaxConnections(),
            knnScalarQuantizedVectorsFormatParams.getBeamWidth(),
            1,
            knnScalarQuantizedVectorsFormatParams.getBits(),
            knnScalarQuantizedVectorsFormatParams.isCompressFlag(),
            knnScalarQuantizedVectorsFormatParams.getConfidenceInterval(),
            null
        );
    }

    @Override
    protected Function<KNNScalarQuantizedVectorsFormatParams, KnnVectorsFormat> getScalarQuantizedVectorsFormatSupplier() {
        return scalarQuantizedVectorsFormatSupplier;
    }
}
```